concrete/src/Package/Package.php

```diff
@@ -781,48 +778,67 @@ public function getNamespace($withLeadingBacksalsh = false)
     }
 
     /**
-     * Create a entity manager used for the package installation,
-     * update and unistall process.
+     * Create a entity manager used for the package installation, update and unistall process.
      *
-     * @return \Doctrine\ORM\EntityManager
+     * @return \Doctrine\ORM\EntityManager|null
      */
     public function getPackageEntityManager()
     {
-        $providerFactory = new PackageProviderFactory($this->app, $this);
-        $provider = $providerFactory->getEntityManagerProvider();
-        $drivers = $provider->getDrivers();
-        if (count($drivers)) {
-            $config = Setup::createConfiguration(true, $this->app->make('config')->get('database.proxy_classes'));
-            $driverImpl = new MappingDriverChain();
-            $coreDriver = new CoreDriver($this->app);
-
-            // Add all the installed packages so that the new package could potentially extend packages that are already
-            // installed
-            $packages = $this->app->make(PackageService::class)->getInstalledList();
-            foreach($packages as $package) {
-                $existingProviderFactory = new PackageProviderFactory($this->app, $package->getController());
-                $existingProvider = $existingProviderFactory->getEntityManagerProvider();
-                $existingDrivers = $existingProvider->getDrivers();
-                if (count($existingDrivers)) {
-                    foreach($existingDrivers as $existingDriver) {
-                        $driverImpl->addDriver($existingDriver->getDriver(), $existingDriver->getNamespace());
+        $result = null;
+        if (!$this instanceof NoDefaultEntityManagerInterface) {
+            $providerFactory = new PackageProviderFactory($this->app, $this);
```

Comment:
This should probably be overridable `$providerFactory = $this->app->make(PackageProviderFactory::class, [$this]);`

Reply:
I just kept the previous code (see https://github.com/concrete5/concrete5/pull/6037/files/0de210b1a545c1ea7d0b8364d556d26df421a19d?w=1#diff-2fdc1f2cb748756db77f4cbcfe01a3d9L789 for an easier view of these edits).